File: backend/src/main/java/az/project/VocabularyBuilder/tictactoe/dto/GameHistoryDto.java
package az.project.VocabularyBuilder.tictactoe.dto;

import az.project.VocabularyBuilder.tictactoe.domain.GameResult;
import az.project.VocabularyBuilder.tictactoe.domain.GameType;

import java.time.LocalDateTime;
import java.util.UUID;

public record GameHistoryDto(
        UUID sessionUuid,
        LocalDateTime datePlayed,
        GameType gameType,
        GameResult result,
        int movesCount,
        int durationSeconds
) {
}

File: frontend/src/app/shared/models/gameHistoryDto.ts
import { GameResult } from "../enums/gameResult";
import { GameType } from "../enums/gameType";

export interface GameHistoryDto {
    sessionUuid: string;

    datePlayed: string;

    gameType: GameType;

    result: GameResult;

    movesCount: number;

    durationSeconds: number;
}